### PR TITLE
Add notes about recent changes on KnotInfo

### PR DESCRIPTION
**Pull request by soehms**
### 📚 Description

This is a follow up of #33969, which used the Khovanov polynomial KnotInfo data in a new conversion method. Unfortunately, this data has since been removed from the website (but not from the database).

This PR merely adds appropriate notes to the documentation of the KnotInfo interface module.

Since the missing data may cause confusion, it would be great if the notes will be included in release 9.8.

### 📝 Checklist

- [x] I have made sure that the title is self-explanatory and the description concisely explains the PR.
- [ ] I have linked an issue or discussion.
- [ ] I have created tests covering the changes.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/sage/knots/knotinfo.py [hunk 1]
```diff
@@ -1714,9 +1720,23 @@ def khovanov_polynomial(self, var1='q', var2='t', base_ring=ZZ, original=False):
...
-            The Khovanov polynomial given in KnotInfo corresponds to the mirror
-            image of the given knot for a `list of 140 exceptions
-            <https://raw.githubusercontent.com/soehms/database_knotinfo/main/hints/list_of_mirrored_khovanov_polynonmial.txt>`__.
+            The data used here were calculated with the program
+            `KhoHo <https://github.com/AShumakovitch/KhoHo>`__. They are no longer
+            visible on the website as of October 30, 2022. Instead, data
+            computed with `KnotJob <https://www.maths.dur.ac.uk/users/dirk.schuetz/knotjob.html>`__
+            are now displayed. The latter program is more accurate in terms of
+            orientation and reflection as it is based on `PD` code.
+
```
Comment (mkoeppe) [inline src/sage/knots/knotinfo.py:1729]: The markup of PD and KhoHo is not right, I think. Should be double backticks or no backticks

Comment (soehms) [reply src/sage/knots/knotinfo.py:1729]: Of course! Many thanks!

BTW: Do we have a replacement for the documentation badge on the Trac ticket header? That has been very convenient!

Comment (soehms) [reply src/sage/knots/knotinfo.py:1729]: > BTW: Do we have a replacement for the documentation badge on the Trac ticket header? That has been very convenient!

Obviously I was to unpationed!

**Review by mkoeppe — APPROVED** [review]
LGTM!

**Comment by soehms** [comment]
> LGTM!

Thanks!

**Pull request merged**